### PR TITLE
Improved accessibility of telegram and quality of voice messages

### PR DESCRIPTION
**Pull request by AlexandrKozlovskiy**
In this pull request i want to improve accessibility of telegram. First of all,i optimize information about forwarded message,and fixed problem with virtualViewId,if message is forwarded. Also i added contentDescription to bot menue view,and added expand/collapse accessibility action,to blind users can know,whether this view expanded/collapsed. Also i fixed problem,when in acceptDeclineView we have sound from talkback,which shows,what there is no next/previous node on the screen. Also i added seekbar not only for voice messages,but for video messages too. Also i added improvement of quality of recording of voice for voice and video messages.

And probably other fixes,which i not mentioned.

# known issues,which was before modifying of sources of telegram:

1. Please record video message. After it you can't play it before sending,because we haven't button for this,as for voice message.

I dont know,whether this possible for sighted people,but if no,it should be implemented for video messages too.

2. Please focus on some video message. You will hear usefull information,including duration of this message,e.g 12 seconds. Please press double tab and focus on it again. Duration will be 11 seconds. So,what information is more correct,before playing,or after starting of the playing. When we close this chat,and open it again,duration of the message not changed,but if we open this chat after very long time,problem repeat again. and this for each video message (only video,not voice,message).

### CODE REVIEW

**Comment by GeorgyAliabyev** [comment]
Please accept this pullquest.

File: TMessagesProj/src/main/java/org/telegram/messenger/MediaController.java [hunk 1]
```diff
@@ -3461,16 +3460,16 @@ public void startRecording(int currentAccount, long dialogId, MessageObject repl
...
                     AndroidUtilities.runOnUIThread(() -> {
                         recordStartRunnable = null;
                         NotificationCenter.getInstance(currentAccount).postNotificationName(NotificationCenter.recordStartError, guid);
                     });
                     return;
                 }
 
-                audioRecorder = new AudioRecord(MediaRecorder.AudioSource.DEFAULT, sampleRate, AudioFormat.CHANNEL_IN_MONO, AudioFormat.ENCODING_PCM_16BIT, recordBufferSize);
-                recordStartTime = System.currentTimeMillis();
+                audioRecorder = new AudioRecord(MediaRecorder.AudioSource.VOICE_RECOGNITION, sampleRate, AudioFormat.CHANNEL_IN_MONO, AudioFormat.ENCODING_PCM_16BIT, recordBufferSize);
```
Comment (ctwoon) [inline TMessagesProj/src/main/java/org/telegram/messenger/MediaController.java:3471]: why

**Comment by DraganRatkovich** [comment]
Any news on this pull request? I still see accessibility issues on Android and voice quality is still the same as whatsapp.
@DrKLO Really wondering if Telegram will ever replace voice recording quality on Android with something better like it is now in video message?

**Comment by DraganRatkovich** [comment]
As always, Telegram doesn't even care about accessibility, let alone other things they need to improve. Anyway, @AlexandrKozlovskiy do you have the compiled .apk from those commits you pushed?

**Comment by AlexandrKozlovskiy** [comment]
@DraganRatkovich I see,you probably deleted your latest coment. I can compile tg for you,but all this changes and more you will see in [nekogram](https://gitlab.com/Nekogram/nekogram). I cammunicate with @xaxtix in tg,but it seems,i almost haven't any result. So,@DraganRatkovich,probably i can not do anything.

**Comment by DraganRatkovich** [comment]
Hi @AlexandrKozlovskiy

I'm sorry, I did a few cleanups and probably my last one left on this commit was also removed. I've used Nekogram a few times and it's really good in terms of accessibility. Hope you can make more changes there.

**Pull request closed without merging**